### PR TITLE
BUGFIX: Use stable identifier for auto-created child nodes in repair

### PR DESCRIPTION
**Pull request by hlubek**
If auto-created child nodes are added via `node:repair`, the generated identifiers are not the same across different variants. This will cause problems when publishing these node variants later.

This change synchronizes the behavior between `node:repair` and regular node creation.

Running `node:repair` will now check identifiers of auto-created child nodes and adjust them accordingly. This is potentially breaking if a child node was referenced from another node but that should very rarely happen.

NEOS-1783 #close

### CODE REVIEW

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php [hunk 1]
```diff
@@ -963,7 +964,7 @@ public function createNode($name, NodeType $nodeType = null, $identifier = null)
             }
 
             foreach ($nodeType->getAutoCreatedChildNodes() as $childNodeName => $childNodeType) {
-                $childNodeIdentifier = $this->buildAutoCreatedChildNodeIdentifier($childNodeName, $newNode->getIdentifier());
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php]: Shouldn't `buildAutoCreatedChildNodeIdentifier` be removed from `Node` then?

**Comment by kitsunet** [comment]
Looks good, left a comment, but that could also be solved in a separate PR as cleanup.
:+1:

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php (hunk 1 above)
Comment (aertmann) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php]: +

**Comment by hlubek** [comment]
Alright, must be a merge error when I merged this back to the 2.0 branch. Will fix.

**Comment by hlubek** [comment]
Alright, I looked for a solution to fix already broken installations and it seems to be pretty easy to fix in the  same part that checks for missing auto-created child nodes without any further overhead.

**Comment by kitsunet** [comment]
:+1:

**Comment by aertmann** [comment]
Good job. Verified.

One problem is that the updating of the unstable identifiers can potentially be breaking if an identifier from a auto-created node is used or referenced. Another one of those unfortunately situations where both options are bad :/ Personally think that it's better to fix them since it's unlikely people reference those auto-created nodes.

Please provide a description about what it fixes, what to do to fix installations and mention the potentially breaking issue to the PR before merging.

**Comment by hlubek** [comment]
@aertmann I updated the description and would merge this PR if nobody objects

**Pull request merged**